src/components/Field/DatePickerField/DatePickerField.tsx

```diff
@@ -0,0 +1,102 @@
+import { useCallback, useState } from 'react';
+
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import LocalizationProvider from '@mui/lab/LocalizationProvider';
+import MuiDatePicker, { DatePickerProps as MuiDatePickerProps } from '@mui/lab/DatePicker';
+import { TextFieldProps as MuiTextFieldProps } from '@mui/material/TextField';
+import { sentenceCase } from 'change-case';
+import cloneDeep from 'lodash.clonedeep';
+
+import TextField from '../TextField';
+import Typography from '../../Typography';
+
+export type DatePickerProps = { 
+  name: string;
+  label?: string;
+  helperText?: string;
+  textFieldProps?: Partial<MuiTextFieldProps>;
+  error?: string,
+  onChangeCallback?: (date: any, keyboardInputValue?: string | undefined) => void,
+  onBlurCallback?: () => void,
+} & MuiDatePickerProps;
+
+const DatePickerField = ({
+    name,
+    label,
+    helperText,
+    textFieldProps = {},
+    onChangeCallback,
+    onBlurCallback,
+    error,
+}: DatePickerProps ): JSX.Element => {
+  const [value, setValue] = useState<Date | string | null>(null);
+
+  const fieldLabel = label != null ? label : sentenceCase(name);
+
+  const handleChange = onChangeCallback || useCallback((val: Date | string | null) => {
+    setValue(val);
+  }, [setValue]);
+
+  const formatDate = (date: string): string => {
+    const dateArr = date.split('/');
+    // if contractor hasn't entered a year, leave date as is
+    // returning an empty string causes an "invalid date" error
+    // to be displayed and the entered text to be retained
+    if (dateArr.length < 3 || dateArr?.[2] === '') return '';
+    let year = dateArr.pop();
+    if (year && year.length < 4) {
+      // prepend a 2 and then however many zeroes to make it length of 4
```

Comment:
nit for comments, I'd want to use this format of comments `/** */` so that the IDE can better associate